src/Validator.php

```diff
@@ -235,19 +233,37 @@ public static function validateType($typeName)
      * Validates a type to be used as a method parameter type hint.
      *
      * @param string $typeName The name of the type to test.
+     * @param bool $allowClass Allow class type hints or not
      * @return null|string Returns a valid type hint for the type or null if there is no valid type hint.
      */
-    public static function validateTypeHint($typeName)
+    public static function validateTypeHint($typeName, $allowClass = false)
     {
-        $typeHint = null;
+        if (self::isInternalPhpType($typeName)) {
+            return null;
+        }
+
+        //TODO: scalar type hints are avalible in PHP7 only
+        if (self::isInternalPhpType($typeName)) {
+            return null;
+        }
 
         // We currently only support type hints for arrays and DateTimes.
         // Going forward we could support it for generated types. The challenge here are enums as they are actually
         // strings and not class instances and we have no way of determining whether the type is an enum at this point.
         if (substr($typeName, -2) == "[]") {
-            $typeHint = 'array';
+            return 'array';
```

Comment:
Oh! I think I was very tired when wrote this -___-

Some changes are need for usage in [Service.php](https://github.com/wsdl2phpgenerator/wsdl2phpgenerator/pull/255/files#diff-16a921c287a57770d0ab59c0fb771b5dR254). I'll fix other changes now.

P.S. Have some thoughts about enums. They are only thing that stops us from class type hinting now, as I see. I'll create new issue to discuss it when work on this will be done.